grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -585,6 +585,9 @@ class OrderService {
                         if (!product) {
                             throw new ProductException("Unable to locate product with product code ${productCode}")
                         }
+                        if (order.destination.isAccountingRequired() && !product.glAccount) {
```

Comment:
We can leave this for now but this feels like it could be added as a custom validation on order item.